google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/CountErrorsPerInterceptorTask.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.data.v2.stub;
+
+import com.google.cloud.bigtable.stats.StatsRecorderWrapper;
+import com.google.cloud.bigtable.stats.StatsWrapper;
+import com.google.common.collect.ImmutableMap;
+import java.util.Set;
+
+class CountErrorsPerInterceptorTask implements Runnable {
+  private final Set<ConnectionErrorCountInterceptor> interceptors;
+  private final StatsRecorderWrapper statsRecorderWrapper;
+
+  public CountErrorsPerInterceptorTask(
+      Set<ConnectionErrorCountInterceptor> interceptors,
+      ImmutableMap<String, String> builtinAttributes) {
+    this.interceptors = interceptors;
+    this.statsRecorderWrapper = StatsWrapper.createRecorder(null, null, builtinAttributes);
+  }
+
+  @Override
+  public void run() {
+    for (ConnectionErrorCountInterceptor interceptor : interceptors) {
+      int errors = interceptor.getAndResetNumOfErrors();
+      int successes = interceptor.getAndResetNumOfSuccesses();
+      // We avoid keeping track of inactive connections (i.e., without any failed or successful
```

Comment:
will inactive connections get cleaned up from the set? I think we should clean up inactive connections to avoid the set growing unbounded

Reply:
Also chatted with Igor and I don't think we can use WeakHashMap cause it's gonna be accessed from multiple thread. Probably need to use a ConcurrentHashMap here and clean up the stale connections manually :(

Reply:
looks like there is a Collections.synchronizedSet() method that does it! :)